lsp-mode.el

```diff
@@ -3706,73 +3706,94 @@ and the position respectively."
+           (resp (lsp-request-while-no-input "textDocument/completion"
+                                             (plist-put (lsp--text-document-position-params)
+                                                        :context (ht ("triggerKind" 1)))))
+           (items (lsp--sort-completions (cond
+                                          ((seqp resp) resp)
+                                          ((hash-table-p resp) (gethash "items" resp)))))
+           (start-point (point))
+           (result (seq-into (--map (-let [(&hash "label"
+                                                  "insertText" insert-text
+                                                  "sortText" sort-text)
+                                           it]
+                                      (propertize (or label insert-text)
+                                                  'lsp-completion-item it
+                                                  'lsp-sort-text sort-text))
+                                    items)
+                             'list))
+           (bounds (or
+                    (-when-let* ((item (lsp-elt items 0))
+                                 ((&hash "textEdit" text-edit) item)
+                                 ((&hash "range") text-edit))
+                      (lsp--range-to-region range))
+                    (bounds-of-thing-at-point 'symbol))))
+      (list
+       (if bounds (car bounds) (point))
+       (point)
+       (lambda (probe pred action)
          (cond
-          (text-edit
-           (delete-region start-point (point))
-           (lsp--apply-text-edit text-edit))
-          (insert-text
-           (delete-region (- (point) (length candidate)) (point))
-           (insert insert-text)))
-
-         (when (eq insert-text-format 2)
-           (yas-expand-snippet (buffer-substring start-point (point))
-                               start-point
-                               (point)))
-         (when additional-text-edits
-           (lsp--apply-text-edits additional-text-edits)))
-
-       (when lsp-signature-auto-activate
-         (lsp-signature-activate))
-
-       (when (lsp--looking-back-trigger-characters-p trigger-chars)
-         (setq this-command 'self-insert-command))))))
+          ((eq action 'metadata)
+           `(metadata . ((display-sort-function
+                          . (lambda (candidates)
+                              (--sort (string-lessp
+                                       (get-text-property 0 'lsp-sort-text it)
+                                       (get-text-property 0 'lsp-sort-text other))
+                                      candidates)))
+                         (category . lsp-substring))))
+          ((eq (car-safe action) 'boundaries) nil)
+          (t
+           (let* ((str (replace-regexp-in-string
+                        (rx (and bos (1+ (not alnum))))
+                        "" probe))
+                  (candidates (completion-all-completions
```

Comment:
here we should match against filterText property of CompletionItem, not against the label.

Reply:
Done